src/reference/00-Getting-Started/07A-Scopes.md

```diff
@@ -393,3 +375,178 @@ lazy val util = (project in file("util"))
+    },
+    bar in Compile := 1
+  )
+```
+
+`(bar in Test)` is undefined, but due to Rule 2 it will look for `(bar in Runtime)` first,
+which is `Test` configuration's parent, then `(bar in Compile)`, which is its parent.
+
+- Rule 3: For Task axis, sbt searches for the given value, and then falls back to `Global`,
+  which is non-task scoped version of the scope.
+
+We can apply Rule 3 to think about the `(scalacOptions in (Compile, console))` example.
+Since `(scalacOptions in (Compile, console))` was not there, Rule 3 says that one of
+the delegates is `(scalacOptions in Compile)`.
+
+These are well defined rules, but you might want to look up quickly what is going on.
+This is where `inspect` can be used.
+
+```
+Hello> inspect projC/compile:console::scalacOptions
+[info] Task: scala.collection.Seq[java.lang.String]
+[info] Description:
+[info]  Options for the Scala compiler.
+[info] Provided by:
+[info]  {file:/Users/xxxx/}projC/compile:scalacOptions
+[info] Defined at:
+[info]  /Users/xxxx/build.sbt:47
+[info] Reverse dependencies:
+[info]  projC/compile:console
+[info]  projC/*:test
+[info] Delegates:
+[info]  projC/compile:console::scalacOptions
+[info]  projC/compile:scalacOptions
+[info]  projC/*:console::scalacOptions
+[info]  projC/*:scalacOptions
+[info]  {.}/compile:console::scalacOptions
+[info]  {.}/compile:scalacOptions
+[info]  {.}/*:console::scalacOptions
+[info]  {.}/*:scalacOptions
+[info]  */compile:console::scalacOptions
+[info]  */compile:scalacOptions
+[info]  */*:console::scalacOptions
+[info]  */*:scalacOptions
+....
+```
+
+See "Provided by". This shows that `projC/compile:console::scalacOptions` is
+actually delegated to `projC/compile:scalacOptions`.
+Next see the "Delegates". This is *all* of the possible delegate candidates
+listed in the order of precedence.
+
+- All the scopes with `projC` scoping on Subproject axis are listed first,
+  then `ThisBuild` (`{.}`), and `Global` (`*`).
+- Within a subproject, scopes with `Compile` scoping on Dependency configuration axis
+  are listed first, then falls back to `Global` (`*`) configuration.
+- Finally, Task axis scoping lists the given task value `console::` and the one without.
+
+Note that scope delegation feels similar to class inheritance in an object-oriented language,
+but there's a difference. In an OO language like Scala if there's a method named
+`drawShape` on a trait `Shape`, its subclasses can override that even when `drawShape` is used
+within the `Shape` trait, which is called dynamic dispatch.
+
+In `build.sbt`, however, if you define a setting in terms of another setting at the build level,
+a key scoped to project level might delegate to the build level, but it will *not* come back.
+Here is an example.
+
+```scala
+lazy val root = (project in file("."))
+  .settings(
+    inThisBuild(List(
+      organization := "com.example",
+      scalaVersion := "2.12.2",
+      version      := scalaVersion.value + "_0.1.0"
+    )),
```

Comment:
this seems to suggest that inThisBuild works/makes sense only in a `settings()` call, while (IIRC) it works anywhere